ui_testing/testcases/basic_tests/test06_orders.py

```diff
@@ -2128,7 +2104,27 @@ def test033_Duplicate_main_order_and_cahange_materiel_type(self):
         self.assertEqual(suborder_data['Article Name'].replace("'", ""), selected_test_plan['article'][0])
         self.assertEqual(suborder_data['Test Units'], test_unit)
         self.assertEqual(suborder_data['Test Plans'], selected_test_plan['testPlanName'])
-        
+
+    def test035_archived_test_unit_shoudnt_display_in_the_order_drop_down_list(self):
+        """
+        Orders: Archived Test unit: Archive Approach: Archived test units shouldn't appear in orders in the drop down list
+        LIMS-3710
+        :return:
+        """
+        testunits, payload = self.test_unit_api.get_all_test_units(limit=20, deleted=1)
+        random_row = random.choice(testunits['testUnits'])
+
+        self.order_page.get_orders_page()
+        self.info('select random order record')
+        orders, payload = self.orders_api.get_all_orders(limit=20)
+        order_row = random.choice(orders['orders'])
+
+        self.orders_page.get_order_edit_page_by_id(id=order_row['id'])
+
+        result = self.order_page.update_suborder(test_units=[random_row['name']], sub_order_index=0)
```

Comment:
Update suborder has no "return" statement in it 
So "result" value  not defined
Also, you need to make assertion that after set test unit , get test unit returns null

Reply:
@0xIslamTaha plesae Islam I wan your help here we have same as this test case and already merged on the master branch so I want here to search on archived test units and it should't appear in the drop down list same as this test case in the roles and permissions section same as here  
        result = self.header_page.set_user_role(user_role=role_random_name)
        self.assertFalse(result, 'no results found ') so this is correct

Reply:
@dianaeldemerdash 
- If I merged the same thing before, then It is my fault and please send it to me and I'm gonna fix it.

- I debugged it and I can see that @ghada-37 already mentioned the correct behavior and here is the details:
As u can see from this [implementation](https://github.com/modeso-open-source/1lims-automation/blob/0a181e3961b74159c0b49ce8dc9d71882b94377e/ui_testing/pages/order_page.py#L399), there is no return value in your call and if u tracked the code u can see that `result` has no value so `self.assertFalse(result)` will always pass.

![image](https://user-images.githubusercontent.com/17981089/79375057-4f269680-7f58-11ea-9088-b17668ee8bd8.png)

- As a suggestion, here u need to create a new function to `check_if_test_unit_existing` or not and it should return false in this case

Reply:
Done